### ISSUE TITLE
pdb4amber strips out nucleic acids in protein/nucleic acid systems if they come before the protein

### ISSUE THREAD

**Issue by drroe** [Amber-MD/pdb4amber#77]
When using `pdb4amber` on a protein/NA-containing PDB where the NA comes before the protein (e.g. 2f5o):
```
pdb4amber -i 2f5o.pdb -o amber.pdb
```
Although amber_renum.txt will contain the NA residues, amber.pdb will not, and it's not in amber_nonprot.pdb either. If the NA comes after the protein (e.g. 6gbm) things seem fine.

**Comment by hainm** [comment]
Ha, odd. I will have a look.

**Comment by dacase** [comment]
That's why these systems are called protein/nucleic acid complexes, and not nucleic acid/protein complexes....  :-)

**Comment by drroe** [comment]
@dacase do you know if the PDB standard says anything about molecule type ordering? I briefly scanned the standard but didn't see anything related to that. It's weird because even though the nucleic acid comes first in 2f5o, it starts as chain B and the protein is chain A (so you'd think it would come after).

**Comment by dacase** [comment]
I meant the ":-)" to indicate that my comment was a joke.  We should have no dependence on molecule type order.

**Comment by drroe** [comment]
Sorry - hope my question/comment didn't come off as snarky. I was just genuinely curious and I don't think I know anyone who knows as much about the PDB standard as you do (the only reason cpptraj conforms as well as it does to the standard are from years of your recommendations!)

**Comment by drroe** [comment]
@hainm it seems that the GIT version of pdb4amber works properly - so maybe the issue is confined to AmberTools 19.

**Comment by hainm** [comment]
Seems fine with AT19 binary (via conda install) on Linux:

![pdb](https://user-images.githubusercontent.com/4451957/64366545-77264280-cfe4-11e9-91e1-293b83b04066.png)

**Comment by hainm** [comment]
> i.e. 6gbm does not work in AT19 as well

Seems fine too
![6gbm](https://user-images.githubusercontent.com/4451957/64366850-1c411b00-cfe5-11e9-90f7-2340f76fc205.png)

**Comment by drroe** [comment]
OK - replace user, try again.

The issue was I had a script that was churning through tons of PDBs and building systems. In this script I had an errant `-p` (keep only protein residues). It was originally there for stripping out heteroatoms that I might not have parameters for. When I expanded to nucleic acid systems I forgot about it. When I tested to try and reproduce with GIT I was not using the `-p` flag, which is why it appeared fine. Sorry for all the hand-waving.

It might be useful to have a `-n` option for systems like this to keep nucleic acid residues as well (while still omitting any heteroatoms). Anyway, problem was me so I'll close this now.
